nexus/src/app/background/tasks/instance_watcher.rs

```diff
@@ -377,15 +377,23 @@ impl BackgroundTask for InstanceWatcher {
     ) -> BoxFuture<'a, serde_json::Value> {
         async {
             let mut tasks = tokio::task::JoinSet::new();
-            let mut paginator = Paginator::new(MAX_SLED_AGENTS);
-            let mk_client = |sled: &Sled| {
-                nexus_networking::sled_client_from_address(
-                    sled.id(),
-                    sled.address(),
-                    &opctx.log,
-                )
-            };
+            let mut paginator = Paginator::new(MAX_CONCURRENT_CHECKS);
```

Comment:
I feel like somewhere there ought to be a comment explaining that the batch size is used as a coarse way to limit concurrency.  Maybe here?

I just noticed while looking at this that this isn't the same as targeting a concurrency level of MAX_CONCURRENT_CHECKS because we'll start 16, then wait until they finish, then start another 16, then wait, ... etc.  So the average concurrency will be somewhat less.  This is fine but I hadn't appreciated it when we talked about it.

Reply:
Yes, that's correct. That's an advantage of using the Tokio semaphore for the concurrency limit rather than the DB query...perhaps we should rework this to use a semaphore _and_ make the query batch size smaller...